src/protagonist/API/Features/Customer/Requests/AssetUpdater.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using API.Features.Customer.Validation;
+using DLCS.Core.Collections;
+using DLCS.Model;
+using DLCS.Model.Assets;
+using DLCS.Repository;
+using DLCS.Repository.Assets;
+using Hydra.Collections;
+using Hydra.Model;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json.Linq;
+
+namespace API.Features.Customer.Requests;
+
+public interface IAssetUpdater
+{
+    /// <summary>
+    /// Updates assets with new values from all image requests
+    /// </summary>
+    public Task<List<Asset>> UpdateAssets(UpdateAllImages request, CancellationToken cancellationToken);
+}
+
+public class AssetUpdater(DlcsContext dlcsContext) : IAssetUpdater
+{
+    public async Task<List<Asset>> UpdateAssets(UpdateAllImages request, CancellationToken cancellationToken = default)
+    {
+        var assetIds = ImageIdListValidation.ValidateRequest(request.HydraUpdate.Members!.Select(m => m.Id).ToList(),
+            request.CustomerId);
+
+        var assets = await dlcsContext.Images
```

Comment:
From ticket:

> Writes should be atomic - competing requests that come in at the same time should both be recorded in the database. See [npgsql EF support for arrays](https://www.npgsql.org/efcore/mapping/array.html#operation-translation) and [RFC-019#implementation-details](https://github.com/dlcs/protagonist/blob/main/docs/rfcs/019-presentation-dlcs.md#implementation-detail)

By loading these upfront, competing requests could end up with differing results in DB as the final stored result is based on the state of data at load time. It'd be better to just issue UPDATE requests to DB,.